server/src-lib/Hasura/Eventing/ScheduledTrigger.hs

```diff
@@ -351,221 +179,199 @@ generateScheduleTimes from n cron = take n $ go from
-  cronScheduledEvents <-
-    liftIO . runExceptT $
-      Q.runTx pgpool (Q.ReadCommitted, Just Q.ReadWrite) getPartialCronEvents
-  case cronScheduledEvents of
-    Right partialEvents -> do
-      -- save the locked cron events that have been fetched from the
-      -- database, the events stored here will be unlocked in case a
-      -- graceful shutdown is initiated in midst of processing these events
-      saveLockedEvents (map cepId partialEvents) lockedCronEvents
-      -- The `createdAt` of a cron event is the `created_at` of the cron trigger
-      for_ partialEvents $ \(CronEventPartial id' name st tries createdAt)-> do
-        case Map.lookup name cronTriggersInfo of
-          Nothing ->  logInternalError $
-            err500 Unexpected "could not find cron trigger in cache"
-          Just CronTriggerInfo{..} -> do
-            let webhook = unResolvedWebhook ctiWebhookInfo
-                payload' = fromMaybe J.Null ctiPayload
-                scheduledEvent =
-                    ScheduledEventFull id'
-                                       (Just name)
-                                       st
-                                       tries
-                                       webhook
-                                       payload'
-                                       ctiRetryConf
-                                       ctiHeaders
-                                       ctiComment
-                                       createdAt
-            finally <- runExceptT $
-              runReaderT (processScheduledEvent logEnv pgpool scheduledEvent Cron) (logger, httpMgr)
-            removeEventFromLockedEvents id' lockedCronEvents
-            either logInternalError pure finally
-    Left err -> logInternalError err
+  -- save the locked cron events that have been fetched from the
+  -- database, the events stored here will be unlocked in case a
+  -- graceful shutdown is initiated in midst of processing these events
+  saveLockedEvents (map _ceId cronEvents) lockedCronEvents
+  -- The `createdAt` of a cron event is the `created_at` of the cron trigger
+  for_ cronEvents $ \(CronEvent id' name st _ tries _ _)-> do
+    case Map.lookup name cronTriggersInfo of
+      Nothing ->  logInternalError $
+        err500 Unexpected "could not find cron trigger in cache"
+      Just CronTriggerInfo{..} -> do
+        let webhookUrl = unResolvedWebhook ctiWebhookInfo
+            payload = ScheduledEventWebhookPayload id' (Just name) st
+                      (fromMaybe J.Null ctiPayload) ctiComment
+                      Nothing
+            retryCtx = RetryContext tries ctiRetryConf
+        finally <- runMetadataStorageT $ flip runReaderT (logger, httpMgr) $
+                   processScheduledEvent logEnv id' ctiHeaders retryCtx
+                                         payload webhookUrl Cron
+        removeEventFromLockedEvents id' lockedCronEvents
+        onLeft finally logInternalError
```

Comment:
```suggestion
    CronTriggerInfo{..} <- onNothing (Map.lookup name cronTriggersInfo) $ 
      logInternalError $ err500 Unexpected "could not find cron trigger in cache"
    let webhookUrl = unResolvedWebhook ctiWebhookInfo
        payload = ScheduledEventWebhookPayload id' (Just name) st
                  (fromMaybe J.Null ctiPayload) ctiComment
                  Nothing
        retryCtx = RetryContext tries ctiRetryConf
    finally <- runMetadataStorageT $ flip runReaderT (logger, httpMgr) $
               processScheduledEvent logEnv id' ctiHeaders retryCtx
                                     payload webhookUrl Cron
    removeEventFromLockedEvents id' lockedCronEvents
    onLeft finally logInternalError
```

Reply:
(This suggestion didn't actually work out, and so I have reverted it on the branch.)